### ISSUE TITLE
MusicFX Crash

### ISSUE THREAD

**Issue by shaydzmi** [Magisk-Modules-Alt-Repo/audio-misc-settings#14]
After installing it, the phone equalizer won't work anymore (musicFX app).
How to uninstall it? I did but the number of steps in media volume to 100 steps is still there.

**Comment by yzyhk904** [comment]
Do you use crDroid or BlissRom? They use a special volume step feature so that you can change volume steps in crDroid (or Bliss) settings (volume levels in the sound section). This feature keeps the previous number of volume steps in some place separately in the settings DB.

Probably, you need an extra reboot for reverting the number of media volume steps. If this fails, please use the above crDroid (or Bliss) settings to change the number to your favorite one, i.e., 15 or 30.

**Comment by shaydzmi** [comment]
No, I use SparkOS ROM, I don't think it has this feature, but this is not
what is important to me, the problem is the system app (MusicFX) which is
an equalizer which I use all of the time is not working anymore. So I want
to uninstall this module completely, in order to get the app to work again.
How can I do this?

**Comment by yzyhk904** [comment]
You can uninstall this module in the Magisk manager as usual. It isn't necessary to reinstall MusicFX at all.
You can use MusicFX again after uninstalling this module in the Magisk manager and rebooting.

This module temporally disables the effects framework itself interfacing to equalizers. Please read its README.md. I think disabling the effects framework itself is the best equalizing.

**Comment by shaydzmi** [comment]
Ok, after the uninstallation, everything goes back to normal, no need for further file copying?

**Comment by yzyhk904** [comment]
So what's the problem? You can change the volume level as before, can't you?
My module changes temporally system properties and libraries, and doesn't any harmful thing.

**Comment by shaydzmi** [comment]
I didn't mean to imply that your module is causing any harm.
 I simply wanted to confirm that if it were to be uninstalled, everything
would return to its state prior to installation.
I've been experiencing issues since I installed it, and I'm curious to know
whether it is the source of the problem or merely a coincidence.
Thank you.

**Comment by omsound** [comment]
Perhaps my comment will be late, I also use the module, I also have a built-in MusicFX equalizer and another Misound, I changed these two lines in the system.prop file: 
ro.audio.ignore_effects=false
ro.audio.spatializer_enabled=true 
The module really improved the sound quality! 
 @yzyhk904 Thanks for the development!

**Comment by yzyhk904** [comment]
> Perhaps my comment will be late, I also use the module, I also have a built-in MusicFX equalizer and another Misound, I changed these two lines in the system.prop file: ro.audio.ignore_effects=false ro.audio.spatializer_enabled=true The module really improved the sound quality! @yzyhk904 Thanks for the development!

Don't change the two lines!. Please consider because this degrades audio quality significantly by increasing jitter (making all music very foggy and vague sound).

I think disabling both of the effects framework and the spatial effect framework is the best and only equalizing for hi-fi audio.

**Comment by omsound** [comment]
It is important to me that the MusicFX equalizer and MiSound work! 
The first is used in the Yandex Music application, the second is a multifunctional equalizer built into MIUI with Dolby Atmos, I have a phone with stereo speakers. 
After I installed the mod, the sound became much cleaner, there was more bass in the headphones! 

What lines should I change to make the equalizers work? 
And how do I disable the effects framework and the spatial effects framework?

**Comment by yzyhk904** [comment]
You cannot enable equalizers at all. My module automatically disable the effects framework and the spatial effect framework both intendedly. I think they are harmful for hi-fi audio use. If you prefer equalizers, I recommend using internal ones in popular hi-fi music players. System wide equalizers including spatial effects are especially harmful because of increasing jitter very much.

If you change the lines, it isn't my module but it's yours. Have you ever used [Audio Jitter Silencer](https://github.com/Magisk-Modules-Alt-Repo/audio-jitter-silencer)?  If you use both "audio misc. settings" and this, you can hear the difference between enabling and disabling the frameworks. especially on bluetooth earphones.

**Comment by omsound** [comment]
And yet I turned off the equalizers, so I need it. 
Not all music apps have a built-in equalizer, such as online music subscription services that I use. 
The built-in equalizer is used, and without it the sound is flat and disgusting. 
Regarding the effects framework, did I understand correctly that they are already disabled and nothing needs to be done? 
I have also installed your DRC remover module, it works fine, thank you!

**Comment by omsound** [comment]
I also want to say about Spatial sound with head tracking support, this is just the line: ro.audio.spatializer_enabled=false
 Devices that support this are already being sold, for example, xiaomi buds 3t pro headphones, I considered them for purchase and my phone also supports this function. 
So it may be worth pointing out in the description of your module that when installing the "Spatial Sound with head tracking support" function will not work so that people know and can change the line if necessary

**Comment by yzyhk904** [comment]
> Regarding the effects framework, did I understand correctly that they are already disabled and nothing needs to be done? I have also installed your DRC remover module, it works fine, thank you!

It isn't true. The framework generates large jitter to distort digital music even it is enabled only and not used by any equalizer. This is the cause of higher audio quality of UAPP, Neutron and other hi-fi players having a feature bypassing the OS mixer.

> I also want to say about Spatial sound with head tracking support, this is just the line: ro.audio.spatializer_enabled=false Devices that support this are already being sold, for example, xiaomi buds 3t pro headphones, I considered them for purchase and my phone also supports this function. So it may be worth pointing out in the description of your module that when installing the "Spatial Sound with head tracking support" function will not work so that people know and can change the line if necessary

I have already written that  "spatial audio" feature was harmful for "hi-fi" use because of increasing jitter significantly. My module also clears out "AUDIO_OUTPUT_FLAG_SPATIALIZER" flag in a bluetooth audio policy file and disables the Moto Dolby app.

If you prefer "spatial audio", you can enable it in your module after forking or cloning from my module source.

If you like "hi-fi" audio, try my another module [hifi maximizer](https://github.com/yzyhk904/hifi-maximizer-mod) and [Audio Jitter Silencer](https://github.com/Magisk-Modules-Alt-Repo/audio-jitter-silencer) simultaneously. The former is a special version of this module for hi-fi audio enthusiasts which disables cameras, thermal controls, the Doze, Logd's, SELinux, etc..

**Comment by omsound** [comment]
Hello!
I don't use spatial sound, I set the ro.audio.spatializer_enabled setting to false. 
UAPP, Neutron will try these players, thank you.
 I understood about the sound shaking and hi-fi, I'll read about the modules and try if it suits me.
 I'm not such a fan of hi fi sound, of course, to turn everything off in the phone)) 
Thanks for the hints! Have a nice day!

**Comment by omsound** [comment]
Please tell me the ADB command to see the status of DRC disabled or enabled, if there is such a command for the console of course

**Comment by yzyhk904** [comment]
There isn't such a specific command, but you can get the status.

1. dumpsys media.audio_policy | awk ' 
        /^ Config source: / {
            print $3
        }' 

You can get the right audio config. file name.

2. more the_file_name_you_got

Search "speaker_drc_enabled", and check its value. DRC is enabled for all audio outputs if the value is "true".
The name of "speaker_drc_enabled" is historical and doesn't mean for speakers only but does for all audio outputs. See README.md.

"Absolute Volume" isn't related to DRC at all.

**Comment by omsound** [comment]
Thanks a lot, I'll try. I have installed your module DRC remover, but I don't understand whether it works or not. 
 I thought that absolute volume is DRC, but it turns out that it is not, thank you

**Comment by omsound** [comment]
I found and checked the configuration file, found the following lines: 


    <globalConfiguration speaker_drc_enabled="false" call_screen_mode_supported="true"/>

This means everything is fine, your module is working, I understood correctlyThis means everything is fineIt means everything is in the softwareIt means everything is fine,, your module is working, am I right?

**Comment by yzyhk904** [comment]
This place is for "Audio Misc. Settings". Please make new issue at the "drc-remover" issues if you have any.
And again see README.md!
